### PR TITLE
docs: clarify that 'second-instance' is only emitted when the second instance calls `app.requestSingleInstanceLock()`

### PR DESCRIPTION
**Pull request by miniak**
#### Description of Change
Address https://github.com/electron/electron/issues/17604

#### Checklist
- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] relevant documentation is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)

#### Release Notes
Notes: no-notes

### CODE REVIEW

**Comment by miniak** [comment]
cc @robatwilliams

File: docs/api/app.md [hunk 1]
```diff
@@ -408,6 +408,8 @@ non-minimized.
 This event is guaranteed to be emitted after the `ready` event of `app`
 gets emitted.
 
+**Note:** It is only emitted when the second instance calls `app.requestSingleInstanceLock()`.
```
Comment (ppontes) [inline docs/api/app.md]: Suggestion: this seems to be relevant enough information to be part of the main paragraph. Currently, the first sentence is incomplete.

Comment (miniak) [reply docs/api/app.md]: please check again

**Comment by codebytere** [comment]
@miniak can you please label the backport targets for this PR?

**Pull request merged**